### DISCUSSION TITLE
[namespace id ] how to setup it before i startup my app

### DISCUSSION THREAD

**Discussion by sangjiexun** [eclipse-milo/milo#1029] [General]
[namespace id ] how to setup it before i startup my app
![image](https://user-images.githubusercontent.com/59449516/184804371-5ab3be96-0d1c-4e2f-a847-e49282da9b08.png)

**Comment by kevinherron** [comment]
I’m not really sure what you’re asking. Are you using Milo to implement a client or server?

The namespace index will just depend on where the URI of the namespace that Node belongs to resides in the server namespace table.

**Reply by sangjiexun** [reply]
Thanks for reply,The URI Number How to self-define for OPCUA Server via Milo

**Reply by kevinherron** [reply]
You need to have an implementation of Namespace to provide Nodes in that Namespace.

The easiest way is to use `ManagedNamespace` or `ManagedNamespaceWithLifecycle`. See `ExampleNamespace` for how these are used. These will register the URI with the server Namespace Table on startup for you. The index is assigned in the order they are registered. If you want a namespace at some high index number you need to register other namespaces in front of it.

You can also just implement `Namespace` yourself, using whatever index and URI you'd like, but you are then responsible for making sure the URI appears at the correct index in the server Namespace Table, and you must keep in mind the Namespace Table (array) cannot have "gaps" or null values, you can't have a namespace index 99 without having URIs registered for all the preceding indices.
